docs/LSPApi.md

```diff
@@ -511,6 +511,53 @@ public class MyLSPSemanticTokensFeature extends LSPSemanticTokensFeature {
 }
 ```
 
+### Semantic Tokens File View Providers
+
+LSP4IJ can help incorporate a file's semantic tokens information so that it's readily available and can be used to
+implement specific behavior based on whether an element is a declaration or reference, string or numeric literal,
+comment, etc., based on the reported semantic tokens. This is implemented via a custom `FileViewProviderFactory` and
+`FileViewProvider`. LSP4IJ includes default implementations of these for TextMate files which otherwise lack PSI element
+structure, and it provides a simple way for other file types to gain access to the same features.
+
+#### LSPSemanticTokensFileViewProviderFactory
+
+Most files use a `SingleRootFileViewProvider`, and those that do can use `LSPSemanticTokensFileViewProviderFactory`.
+If specialized behavior is needed, `LSPSemanticTokensFileViewProviderFactory` subclassed and and implemented.
+
+If the custom LSP integration's files are based on a specific language ID, the factory should be registered in 
+`plugin.xml` using `language.fileViewProviderFactory`. It its files are not based on specific language ID, it should be

```

Comment:
Also, I think the correct should be `lang.fileViewProviderFactory` right?

Reply:
Again, yup. I'm actually adding concrete examples to those sections now.

Reply:
@ericdallo I've made a pretty significant pass on the docs. Please let me know if you see any further issues, and please make note of the added qualification about needing to enable the feature flag for semantic tokens file view providers in custom LS integrations.